llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7972,6 +7972,13 @@ Instruction *InstCombinerImpl::visitFCmpInst(FCmpInst &I) {
   Constant *RHSC;
   if (match(Op0, m_Instruction(LHSI)) && match(Op1, m_Constant(RHSC))) {
     switch (LHSI->getOpcode()) {
+    case Instruction::Select:
+      if ((Pred == FCmpInst::FCMP_UEQ || Pred == FCmpInst::FCMP_OEQ ||
```

Comment:
Alive2: https://alive2.llvm.org/ce/z/7DMRCy

Reply:
could I use this `computeKnownFPClass(Op1).isKnownNever(FPClassTest::fcNan | FPClassTest::fcInf | FPClassTest::fcNormal | FPClassTest::fcSubnormal)`?

Reply:
No. All you need is `m_AnyZero`.

Reply:
I'm concerned this (half nofpclass(inf nan norm sub) %y) might cause test failures.

Reply:
> I'm concerned this (half nofpclass(inf nan norm sub) %y) might cause test failures.

It is for alive2 proof, but not for tests.